DOCS-6454 Add Understanding How Synthetic Test Retries Determine Monitor Status Guide (#20210)
DataDog/documentation: content/en/synthetics/guide/synthetic-test-retries-monitor-status.md

---

> title: Understanding How Synthetic Test Retries Determine Monitor Status

Author: ```suggestion
title: Understand How Synthetic Test Retries Determine Monitor Status
```

Author: Yes, I remembered that we want to avoid using gerunds in the guide titles 😅

kind: guide
description: Learn how a Synthetic test retry affects the associated monitor status.
further_reading:
- link: "/synthetics/guide/synthetic-test-monitors/"
  tag: "Documentation"
  text: "Learn about Synthetic test monitors"
- link: "/continuous_testing/explorer/search_runs/"
  tag: "Documentation"
  text: "Learn about Synthetic test runs"
---

## Overview

Synthetic tests can be retried when, or if, a [test run][1] fails. If you have configured a test to be retried on failures, this is a _fast retry_, and the failed test run results appear in the [Synthetic Monitoring & Continuous Testing Explorer][2].

{{< img src="synthetics/guide/synthetics_test_retries/failed_test_runs.png" alt="Failed test runs in the Synthetic Monitoring & Continuous Testing Explorer" style="width:100%;">}}

For more information about monitors associated with your Synthetic tests, see [Using Synthetic Test Monitors][3].

Fast retry results are used in the local group evaluation, but are ignored in the total group evaluation. 

Local Group Evaluation

> : This is the definition for a local group eval.

Author: 🤔💭🙇🏻
```suggestion
: This is the definition for a local group eval.
```

Total Group Evaluation

> : This is the definition for a total group eval.

Author: 🤔💭🙇🏻
```suggestion
: This is the definition for a total group eval.
```

A run that is still failing after it has reached the maximum number of retries is considered final, and this final result is taken into account in the total group evaluation. 

### Retries that overlap with other test runs

In this example, a Synthetic test is scheduled to run every three minutes, and has a retry configured to a maximum of two times with a delay of two minutes.  

{{< img src="synthetics/guide/synthetics_test_retries/diagram_tobereplaced.png" alt="A test run which was retried twice and failed on all retries, evaluated as a local group and as a total group" style="width:100%;">}}

> The alerting evaluator only takes the final retry into account for the total group evaluation. Depending on what you set for the `minFailureDuration` and `minLocationsFailed` parameters, this example may not result in a failed evaluation for the monitor, but instead reveal itself as inconsistencies in the (test or retry?) scheduling.

Author: 🤔💭🙇🏻

```suggestion
The alerting evaluator only takes the final retry into account for the total group evaluation. Depending on what you set for the `minFailureDuration` and `minLocationsFailed` parameters, this example may not result in a failed evaluation for the monitor, but instead reveal itself as inconsistencies in the (test or retry?) scheduling.
```

---
title: Understanding How Synthetic Test Retries Determine Monitor Status
kind: guide
description: Learn how a Synthetic test retry affects the associated monitor status.
further_reading:
- link: "/synthetics/guide/synthetic-test-monitors/"
  tag: "Documentation"
  text: "Learn about Synthetic test monitors"
- link: "/continuous_testing/explorer/search_runs/"
  tag: "Documentation"
  text: "Learn about Synthetic test runs"
---

## Overview

To reduce alert fatigue, Synthetic tests can be retried when, or if, a test run fails. If you have configured a test to be retried on failures, this is a _fast retry_. 

With a fast retry, Datadog runs a Synthetic test multiple times before transitioning the test's monitor to alert and sending you an notification. For more information about monitors associated with your Synthetic tests, see [Use Synthetic Test Monitors][3].

{{< img src="synthetics/guide/synthetics_test_retries/fast_retries.png" alt="Failed test runs with fast retries" style="width:100%;">}}

## Group evaluations

> While fast retry results are used in the local group evaluation, only the final retry is taken into account in the total group evaluation. The original run and all intermediate retries are discarded from the evaluation.
> 
> Local Group Evaluation
> : Evaluation of the location status.
> 
> Total Group Evaluation
> : Evaluation of the test status.
> 

Reviewer: ```suggestion
Two types of group evaluations determine ... <insert information about what these group evaluations are used for and why users should care about them>.

Local Group Evaluation
: Evaluation of the location status.

Total Group Evaluation
: Evaluation of the test status.

While fast retry results are used in the local group evaluation, only the final retry is taken into account in the total group evaluation. The original run and all intermediate retries are discarded from the evaluation.

```

A run that is still failing after it has reached the maximum number of retries is considered final, and this final result is taken into account in the total group evaluation. 

## Retries that overlap with other test runs

In this example, a Synthetic test is scheduled to run every three minutes, and has a retry configured to a maximum of two times with a delay of two minutes.  

The evaluation only takes the final retry into account for the total group evaluation. 

> When all retries fails:

Reviewer: ```suggestion
When all retries fail:
```

{{< img src="synthetics/guide/synthetics_test_retries/diagram_1.png" alt="A test run which was retried twice and failed on all retries, evaluated as a local group and as a total group" style="width:100%;">}}

> or  when a retry is successful:

Author: ```suggestion
Or when a retry is successful:
```

{{< img src="synthetics/guide/synthetics_test_retries/diagram_2.png" alt="A test run which was retried twice and failed on all retries, evaluated as a local group and as a total group" style="width:100%;">}}

**Note:** Depending on what you set for the `minFailureDuration` and `minLocationsFailed` parameters, you may see different behavior.

## Timestamps

> The timestamp for a final result is when the test was retried at, not the time the test was originally scheduled. Results are considered at the timestamp when the test was started. Due to the test's execution time, there may be a small delay before the results become available for the evaluation.

Reviewer: ```suggestion
The system populates the timestamp for a final result with the time when the test was retried, not the time the test was originally scheduled. Results are considered at the timestamp when the test was started. Due to the test's execution time, there may be a small delay before the results become available for the evaluation.
```
